### PR TITLE
GDA: Dev/citations component override

### PR DESCRIPTION
**Pull request by YextSLu**
Changes:
- Added `CitationsContainer` prop to `<GenerativeDirectAnswer>` to allow custom components to be passed in
- Updated test cases and test-site with new component passed through

J=CLIP-1369
TEST=manual|auto

Ran `npm run build`, `npm run test`, and `npm run wcag` and did manual testing on test-site

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -154,19 +154,20 @@
   },
   "dependencies": {
     "@restart/ui": "^1.0.1",
+    "@swc/core": "^1.7.0",
```
Comment (yen-tt) [inline package.json:157]: what is this for?

File: src/components/GenerativeDirectAnswer.tsx [hunk 2]
```diff
@@ -134,7 +137,12 @@ function Answer(props: AnswerProps) {
...
 }
 
-interface CitationsProps {
+/**
+ * Props for citations component.
+ *
+ * @public
+ */
+export interface CitationsProps {
   gdaResponse: GenerativeDirectAnswerResponse,
```
Comment (yen-tt) [inline src/components/GenerativeDirectAnswer.tsx:146]: add docs for each of the fields now that this interface is public

File: tests/components/GenerativeDirectAnswer.test.tsx [hunk 3]
```diff
@@ -24,6 +24,14 @@ const mockedState: Partial<State> = {
   }
 };
 
+const CustomCitationsComponent = (props: CitationsProps) => {
```
Comment (yen-tt) [inline tests/components/GenerativeDirectAnswer.test.tsx:27]: failed linter check -- can you run `npm run lint` locally?

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@yext/search-ui-react",
-  "version": "1.7.0-beta.4",
+  "version": "1.7.1-beta.4",
```
Comment (yen-tt) [inline package.json:3]: `1.7.0-beta.5`

File: etc/search-ui-react.api.md [hunk 5]
```diff
@@ -347,6 +361,7 @@ export interface GenerativeDirectAnswerCssClasses {
 export interface GenerativeDirectAnswerProps {
     answerHeader?: string | JSX.Element;
     CitationCard?: (props: CitationProps) => JSX.Element | null;
+    CitationsComponent?: (props: CitationsProps) => JSX.Element | null;
```
Comment (yen-tt) [inline etc/search-ui-react.api.md:364]: nit: the `Component` suffix is not really needed as it's implied by its type (e.g. like CitationCard), can we rename this?

File: src/components/GenerativeDirectAnswer.tsx (hunk 2 above)
Comment (YextSLu) [reply src/components/GenerativeDirectAnswer.tsx:147]: Added. Not showing as "outdated" because I edited lines below it

**Review by nbramblett — APPROVED** [review]

**Review by yen-tt — APPROVED** [review]

**Pull request merged**